contracts/sol6/Dao/EmergencyFeeHandler.sol

```diff
@@ -0,0 +1,201 @@
+pragma solidity 0.6.6;
+
+import "../IKyberFeeHandler.sol";
+import "../utils/PermissionGroupsNoModifiers.sol";
+import "../utils/zeppelin/ReentrancyGuard.sol";
+import "../utils/zeppelin/SafeMath.sol";
+import "../utils/Utils5.sol";
+
+contract EmergencyKyberFeeHandler is IKyberFeeHandler, PermissionGroupsNoModifiers, ReentrancyGuard, Utils5 {
+    using SafeMath for uint256;
+
+    uint16 public immutable rewardBps;
+    uint16 public immutable rebateBps;
+    address public kyberNetwork;
+
+    mapping(address => uint256) public feePerPlatformWallet;
+    uint256 public totalPlatformFeeWei; // total balance in the contract that is for platform fee
+    mapping(address => uint256) public rebatePerWallet;
+    uint256 public totalRewardWei;
+
+    event HandleFeeFailed(address[] rebateWallets, uint256[] rebateBpsPerWallet, uint256 feeBRRWei);
+
+    event FeeDistributed(
+        address indexed platformWallet,
+        uint256 platformFeeWei,
+        uint256 rewardWei,
+        uint256 rebateWei,
+        address[] rebateWallets,
+        uint256[] rebatePercentBpsPerWallet,
+        uint256 burnAmtWei
+    );
+
+    event EtherWithdraw(uint256 amount, address sendTo);
+
+    constructor(
+        address admin,
+        address _kyberNetwork,
+        uint16 _rewardBps, 
+        uint16 _rebateBps, 
+        uint16 _burnBps
+    ) public PermissionGroupsNoModifiers(admin) {
+        require(_burnBps + _rewardBps + _rebateBps == BPS, "Bad BRR values");
+        require(_burnBps <= BPS, "burnBps overflow");
+        require(_rewardBps <= BPS, "rewardBps overflow");
+        require(_rebateBps <= BPS, "rebateBps overflow");
+        rewardBps = _rewardBps;
+        rebateBps = _rebateBps;
+        kyberNetwork = _kyberNetwork;
+    }
+
+    modifier onlyKyberNetwork {
+        require(msg.sender == address(kyberNetwork), "only Kyber");
+        _;
+    }
+
+
+    /// @dev handleFees function is called per trade on KyberNetwork. unless the trade is not involving any fees.
+    /// @param rebateWallets a list of rebate wallets that will get rebate for this trade.
+    /// @param rebateBpsPerWallet percentage of rebate for each wallet, out of total rebate.
+    /// @param platformWallet Wallet address that will receive the platfrom fee.
+    /// @param platformFeeWei Fee amount in wei the platfrom wallet is entitled to.
+    function handleFees(
+        address[] calldata rebateWallets,
+        uint256[] calldata rebateBpsPerWallet,
+        address platformWallet,
+        uint256 platformFeeWei
+    ) external payable override onlyKyberNetwork {
+        // handle platform fee
+        feePerPlatformWallet[platformWallet] = feePerPlatformWallet[platformWallet].add(
+            platformFeeWei
+        );
+
+        uint256 feeBRRWei = msg.value.sub(platformFeeWei);
+        totalPlatformFeeWei = totalPlatformFeeWei.add(platformFeeWei);
+        if (feeBRRWei == 0) {
+            // only platform fee paid
```

Comment:
I think it is better if we issue event per function call.
no matter what is feeBRRWei

so we have event data for sure.

Reply:
change as request